plugins/additional/additional.cpp

```diff
@@ -443,27 +478,151 @@ class ToggleButtonComponent : public ComponentBase, public wxEvtHandler
+			xrc.AddProperty(_("pressed"), _("pressed"), XRC_TYPE_BITMAP);
+		}
+		if (!obj->IsNull(_("focus"))) {
+			xrc.AddProperty(_("focus"), _("focus"), XRC_TYPE_BITMAP);
+		}
+		if (!obj->IsNull(_("current"))) {
+			xrc.AddProperty(_("current"), _("current"), XRC_TYPE_BITMAP);
+		}
+		if (!obj->IsNull(_("position"))) {
+			xrc.AddProperty(_("position"), _("position"), XRC_TYPE_TEXT);
+		}
+		if (!obj->IsNull(_("margins"))) {
+			xrc.AddProperty(_("margins"), _("margins"), XRC_TYPE_SIZE);
+		}
+		xrc.AddProperty(_("value"),_("checked"), XRC_TYPE_BOOL);
+		return xrc.GetXrcObject();
+	}
+
+	ticpp::Element* ImportFromXrc(ticpp::Element* xrcObj) override {
+		XrcToXfbFilter filter(xrcObj, _("wxToggleButton"));
+		filter.AddWindowProperties();
+		filter.AddProperty(_("label"),_("label"),XRC_TYPE_TEXT);
+		filter.AddProperty(_("markup"), _("markup"), XRC_TYPE_BOOL);
+		filter.AddProperty(_("bitmap"), _("bitmap"), XRC_TYPE_BITMAP);
+		filter.AddProperty(_("disabled"), _("disabled"), XRC_TYPE_BITMAP);
+		filter.AddProperty(_("pressed"), _("pressed"), XRC_TYPE_BITMAP);
+		filter.AddProperty(_("focus"), _("focus"), XRC_TYPE_BITMAP);
+		filter.AddProperty(_("current"), _("current"), XRC_TYPE_BITMAP);
+		filter.AddProperty(_("position"), _("position"), XRC_TYPE_TEXT);
+		filter.AddProperty(_("margins"), _("margins"), XRC_TYPE_SIZE);
+		filter.AddProperty(_("checked"),_("value"), XRC_TYPE_BOOL);
+		return filter.GetXfbObject();
+	}
+};
+
+class BitmapToggleButtonComponent : public ComponentBase, public wxEvtHandler
+{
+public:
+	wxObject* Create(IObject* obj, wxObject* parent) override {
+		wxBitmapToggleButton* button = new wxBitmapToggleButton((wxWindow *)parent, wxID_ANY,
+			obj->GetPropertyAsBitmap(_("bitmap")),
+			obj->GetPropertyAsPoint(_("pos")),
+			obj->GetPropertyAsSize(_("size")),
+			obj->GetPropertyAsInteger(_("style")) | obj->GetPropertyAsInteger(_("window_style")));
+
+#if wxCHECK_VERSION( 2, 9, 2 )
+		// To stay in sync what the generator templates do apply the markup label here as well
+		if (obj->GetPropertyAsInteger(_("markup")) != 0) {
+			button->SetLabelMarkup(obj->GetPropertyAsString(_("label")));
+		}
+#endif
+
+		if ( !obj->IsNull( _("disabled") ) )
+		{
+			button->SetBitmapDisabled( obj->GetPropertyAsBitmap( _("disabled") ) );
+		}
+
+		if (!obj->IsNull(_("pressed"))) {
+			button->SetBitmapPressed(obj->GetPropertyAsBitmap(_("pressed")));
+		}
+
+		if ( !obj->IsNull( _("focus") ) )
+		{
+			button->SetBitmapFocus( obj->GetPropertyAsBitmap( _("focus") ) );
+		}
+
+		if (!obj->IsNull(_("current"))) {
+			button->SetBitmapCurrent(obj->GetPropertyAsBitmap(_("current")));
+		}
+
+		if (!obj->IsNull(_("position"))) {
+			button->SetBitmapPosition(
+			    static_cast<wxDirection>(obj->GetPropertyAsInteger(_("position"))));
+		}
+
+		if (!obj->IsNull(_("margins"))) {
+			button->SetBitmapMargins(obj->GetPropertyAsSize(_("margins")));
+		}
+
+		button->SetValue( ( obj->GetPropertyAsInteger(_("value")) != 0 ) );
+
```

Comment:
Here too the connect of the wxEVT_COMMAND_TOGGLEBUTTON_CLICKED is missing so that button clicks don't update the value in the model